core/zoe/zoe.chainmail

```diff
@@ -0,0 +1,165 @@
+   * with a swap contract and an array of the issuers for moola and
+   * simoleans. We get back an object that has the user facet for the
+   * zoeInstance and the governing contract facet. `makeContract` is a
+   * function that takes a zoeInstance.
+   */
+  makeInstance(makeContract :Object, issuers :List) -> (UserFacetAndGoverningContract);
+
+  /**
+   * Get the long-lived seatIssuer. The mint associated with the
+   * seatIssuer creates the ERTP payments that represent the right to
+   * claim the payoffs of involvement in a contract.
+   */
+  getSeatIssuer() -> (Issuer);
+
+  /**
+   * Get the long-lived escrowReceiptIssuer. The mint associated
+   * with the escrowReceiptIssuer creates the ERTP payments that
+   * represent proof of escrow and may be passed into a
+   * governingContract method.
+   */
+  getEscrowReceiptIssuer() -> (Issuer);
+
+}
+
+/**
+ * An offer description is an array of OfferDescElements. The possible
+ * rules are 'offerExactly', 'offerAtMost', 'wantExactly', and 
+ * 'wantAtLeast'.
+ */
+struct OfferDescElement ( ) {
+  rule :Rule;
+  amount :Amount;
+}
+
+interface ZoeUserFacet {
+
+  /** 
+   * To escrow, the user must provide an `offerDesc` which is a list of
+   * objects with a rule property (such as 'offerExactly') and an amount
+   * property. The objects in the `offerDesc` must be in the same order
+   * as the issuers array. That is, the amount in index 0 of `offerDesc`
+   * should be an amount for the issuer in index 0 of the issuers array.
+   * `payments` is an array of the actual payments to be escrowed,
+   * following the rules in the offerDesc. If the offerDesc rule is
+   * 'offerExactly' or 'offerAtMost', then a payment matching the amount is
+   * expected. If the offerDesc rule is 'wantAtLeast' or 'wantExactly'
+   * then the payment will be ignored and should be undefined.
+   */
+  escrow (offerDesc :List(OfferDescElement), payments :List(Payment)) -> (EscrowReceiptAndClaimPayoff);
+
+  /**
+   * Returns the zoeInstance's array of issuers. This is helpful in 
+   * the case that the user has forgotten the order.
+   */
+  getIssuers() -> (List);
+
+}
+
+interface ZoeGoverningContractFacet () {
+
+  /** 
+   * Instruct Zoe to try reallocating for the given offerIds.
+   * Reallocation is a matrix (array of arrays) where the rows are the
+   * quantities to be paid to the player who made the offer at the same
+   * index in the offerIds array. The reallocation will only happen if
+   * 'offer safety' and conservation of rights are true, as enforced by Zoe.
+   */
+  reallocate (offerIds :List, reallocation :List(List(Quantity)));
+
+  /** 
+   * Eject the offer, taking the current allocation for that offer and
+   * creating payments to be returned to the user. The payments can be
+   * claimed by the user by taking the `claimPayment` ERTP payment
+   * received after escrowing, calling `unwrap` and calling the `claimPayoff`
```

Comment:
If `claimPayoff` a method or a property? The chainmail puts it in a pass-by-copy struct and therefore declares `claimPayoff` as a property.

Reply:
Ah, the names here aren't quite correct. I just pushed up a fix. The ERTP payment is called `claimPayoff` and the method that is used is called `getPayoff`. The chainmail struct is correct that it is a property. We should probably work on the names